subsys/bluetooth/mesh/prov_device.c

```diff
@@ -281,70 +265,39 @@ static void send_pub_key(void)
 	start_auth();
 }
 
-static void dh_key_gen_complete(void)
-{
-	BT_DBG("DHkey: %s", bt_hex(bt_mesh_prov_link.dhkey, BT_DH_KEY_LEN));
-
-	if (!atomic_test_and_clear_bit(bt_mesh_prov_link.flags, WAIT_DH_KEY) &&
-	    atomic_test_bit(bt_mesh_prov_link.flags, OOB_PUB_KEY)) {
-		send_confirm();
-	} else if (!atomic_test_bit(bt_mesh_prov_link.flags, OOB_PUB_KEY)) {
-		send_pub_key();
-	}
-}
-
-static void prov_dh_key_cb(const uint8_t dhkey[BT_DH_KEY_LEN])
-{
-	BT_DBG("%p", dhkey);
-
-	if (!dhkey) {
-		BT_ERR("DHKey generation failed");
-		prov_fail(PROV_ERR_UNEXP_ERR);
-		return;
-	}
-
-	sys_memcpy_swap(bt_mesh_prov_link.dhkey, dhkey, BT_DH_KEY_LEN);
-
-	dh_key_gen_complete();
-}
-
 static void prov_dh_key_gen(void)
 {
-	const uint8_t *remote_pk;
-	uint8_t remote_pk_le[BT_PUB_KEY_LEN];
+	const uint8_t *remote_pub_key;
+	const uint8_t *remote_priv_key;
 
-	remote_pk = bt_mesh_prov_link.conf_inputs.pub_key_provisioner;
+	remote_pub_key = bt_mesh_prov_link.conf_inputs.pub_key_provisioner;
```

Comment:
Generally I agree with you. However, it will cause significant API change since derivation keys pair has only one id. Private\public keys are represented by one id.
Leave it for NCSDK-10576

Reply:
Not sure what is significant as we will just need to export public key by id.

Reply:
Public key as entity doesn't have its own ID.